server/src/main/java/org/elasticsearch/action/search/CanMatchNodeRequest.java

```diff
@@ -146,6 +152,40 @@ public CanMatchNodeRequest(
         indices = shards.stream().map(Shard::getOriginalIndices).flatMap(Arrays::stream).distinct().toArray(String[]::new);
     }
 
+    private static void collectSignificantTermsBackgroundFilters(
+        Collection<AggregationBuilder> aggregations,
+        List<QueryBuilder> backgroundFilters
+    ) {
+        for (AggregationBuilder aggregation : aggregations) {
+            if (aggregation instanceof SignificantTermsAggregationBuilder) {
+                QueryBuilder backgroundFilter = ((SignificantTermsAggregationBuilder) aggregation).backgroundFilter();
+                backgroundFilters.add(backgroundFilter != null ? backgroundFilter : QueryBuilders.matchAllQuery());
```

Comment:
This is because the default background filter is match_all, correct? It could also be incorporated into the new AggregationBuilder method, if possible. Looks like there's impl details of significant terms leaking out to can match, perhaps.

Reply:
You're totally right, and your other suggestion naturally fixes this.